src/mgr/DaemonServer.cc

```diff
@@ -341,12 +350,53 @@ void DaemonServer::maybe_ready(int32_t osd_id)
   }
 }
 
+void DaemonServer::tick()
+{
+  dout(10) << dendl;
+  send_report();
+
+  schedule_tick_locked(
+    g_conf().get_val<std::chrono::seconds>("mgr_tick_period").count());
+}
+
+// Currently modules do not set health checks in response to events delivered to
+// all modules (e.g. notify) so we do not risk a thundering hurd situation here.
+// if this pattern emerges in the future, this scheduler could be modified to
+// fire after all modules have had a chance to set their health checks.
+void DaemonServer::schedule_tick_locked(double delay_sec)
+{
+  if (tick_event) {
```

Comment:
Be nice to have an assert on lock.is_locked_by_me at the start of this function

Reply:
done